docs/reference/modules/discovery/fault-detection.asciidoc

```diff
@@ -31,3 +31,297 @@ The elected master node will also remove nodes from the cluster if nodes are una
+master logs a message like this (with line breaks added for clarity):
+
+```
+[2020-01-29T10:20:35,509][INFO ][o.e.c.s.MasterService    ]
+    [es-master-1] node-left[
+        {es-master-2}{1vSx_i__QCOjVyfA9kYEkg}{q78rb3MsRDyCcweeygxeiA}{10.244.35.101}{10.244.35.101:9300}{m}
+            disconnected,
+        {es-data2-5}{KZMiSjy3QRiJevGQokB-zQ}{stIE-HcITQqLuCbAEgbCiA}{10.244.2.213}{10.244.2.213:9300}{d}
+            disconnected],
+        term: 469, version: 446673, ...
+```
+
+This message says that the `MasterService` on the elected master
+(`es-master-1`) is processing a `node-left` task, and it lists the nodes that
+are being removed and the reasons for their removal. You will also see
+similar-sounding messages on all other nodes like this:
+
+```
+[2020-01-29T10:20:48,985][INFO ][o.e.c.s.ClusterApplierService]
+    [es-master-1] removed {
+        {es-master-2}{1vSx_i__QCOjVyfA9kYEkg}{q78rb3MsRDyCcweeygxeiA}{10.244.35.101}{10.244.35.101:9300}{m}{xpack.installed=true},
+        {es-data2-5}{KZMiSjy3QRiJevGQokB-zQ}{stIE-HcITQqLuCbAEgbCiA}{10.244.2.213}{10.244.2.213:9300}{d}{xpack.installed=true}},
+    term: 469, version: 446676, reason: Publication{term=469, version=446676}
+```
+
+{es} emits this message on every node when applying a cluster state that
+removes one or more nodes from the cluster. There's less useful information
+here, and it's better to focus on the one from the `MasterService` which is
+only emitted on the elected master. If you don't see the messages from the
+`MasterService` then check you're looking at the logs from the elected master
+covering the correct time period and that logging is enabled at `INFO` level.
+
+Nodes will also log a message containing the string `master node changed`
+whenever they start or stop following the elected master. You can use these
+messages to determine each node's view of the state of the master over time.
+
+If a node restarts then it will leave the cluster and then join the cluster
+again. When it rejoins, the `MasterService` will log that it is processing a
+`node-join` task. You can tell from the master logs that the node was restarted
+because the `node-join` message will indicate that it is
+`joining after restart`. In older versions you can also determine that a node
+restarted by looking at the second "ephemeral" ID in the `node-left` and
+subsequent `node-join` messages: this ID is different each time the node starts
+up. If a node is unexpectedly restarting then you'll need to look at the node's
+own logs to see why it is shutting down.
+
+If the node did not restart then you should look at the reason for its
+departure in the `node-left` message, which is reported after each node. There
+are three possible reasons:
+
+* `disconnected`: the connection from the master node to the removed node was
+closed.
+
+* `lagging`: the master published a cluster state update but the removed node
+did not apply it within the permitted timeout. By default this timeout is 2
+minutes.
```

Comment:
This one is more questionable. It's not exactly recommended to change it but it might make sense in some cases. I'll link to the docs about those settings.